skfuzzy/control/fuzzyvariable.py

```diff
@@ -44,12 +82,30 @@ def __init__(self, universe, label):
             Unique name of the universe variable, e.g., 'food' or 'velocity'.
         """
         self.universe = np.asarray(universe)
-        self.active = None
-        self.mf = OrderedDict()
         self.label = label
-        self.connections = OrderedDict()
+        self.defuzzy_method = defuzzy_method
+        self.adjectives = OrderedDict()
+
         self._id = id(self)
-        self.output = None
+        self._crisp_value_accessed = False
+
+        class _NotGenerator(object):
```

Comment:
Nice approach! Should we force users to conform to exactly 'NOT-' as a prefix, or consider matching a real expression that doesn't care about case, and accepts whitespace or a dash as a separator? I'd prefer to type `variable['not mylabel']` over `variable['NOT-mylabel']`. Perhaps a little magic would be acceptable here.